src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs

```diff
@@ -818,43 +817,31 @@ public override int GetHashCode()
             => Win32ProgramEqualityComparer.Default.GetHashCode(this);
 
         public override bool Equals(object obj)
-            => obj is Win32Program win && Win32ProgramEqualityComparer.Default.Equals(this, win);
+            => obj is Win32Program win32Program && Win32ProgramEqualityComparer.Default.Equals(this, win32Program);
 
         private class Win32ProgramEqualityComparer : IEqualityComparer<Win32Program>
         {
             public static readonly Win32ProgramEqualityComparer Default = new Win32ProgramEqualityComparer();
 
-            public bool Equals(Win32Program app1, Win32Program app2)
-                => app1 != null
-                   && app2 != null
-                   && (app1.Name?.ToUpperInvariant(), app1.ExecutableName?.ToUpperInvariant(), app1.FullPath?.ToUpperInvariant())
-                    .Equals((app2.Name?.ToUpperInvariant(), app2.ExecutableName?.ToUpperInvariant(), app2.FullPath?.ToUpperInvariant()));
-
-            public int GetHashCode(Win32Program obj)
-                => (obj.Name?.ToUpperInvariant(), obj.ExecutableName?.ToUpperInvariant(), obj.FullPath?.ToUpperInvariant()).GetHashCode();
-        }
-
-        private class RemoveDuplicatesComparer : IEqualityComparer<Win32Program>
-        {
-            public static readonly RemoveDuplicatesComparer Default = new RemoveDuplicatesComparer();
-
             public bool Equals(Win32Program app1, Win32Program app2)
             {
-                if (app1 == null || app2 == null)
+                if (app1 == null && app2 == null)
                 {
-                    return false;
+                    return true;
                 }
 
-                return (app1.FullPath.ToUpperInvariant(), app1.HasArguments ? app1.Arguments.ToUpperInvariant() : null)
-                    .Equals((app2.FullPath.ToUpperInvariant(), app2.HasArguments ? app2.Arguments.ToUpperInvariant() : null));
+                return app1 != null
+                       && app2 != null
+                       && (app1.Name?.ToUpperInvariant(), app1.FullPath?.ToUpperInvariant())
+                       .Equals((app2.Name?.ToUpperInvariant(), app2.FullPath?.ToUpperInvariant()));
```

Comment:
Can you elaborate on why ExecutableName is removed here?

Reply:
As executablename is always set to the last part of the FullPath!

Reply:
It's not always true. For example, `LnkProgram`:

![image](https://user-images.githubusercontent.com/17161067/119821489-4d6d2780-befb-11eb-8f03-7b65c5a47de8.png)

Also, I found one pair of programs on my machine that have the same FullPath and different ExecutalbeName. I would suggest to return `ExecutableName`.

Reply:
Should we not expand the executable name (so you can also use if for the direct exe match), I don't think anyone wants to search for that name?

Reply:
Let's concentrate on speeding it up without changing behavior where it is possible.